management-api-agent-common/src/main/java/io/k8ssandra/metrics/http/NettyMetricsHttpServer.java

```diff
@@ -0,0 +1,39 @@
+package io.k8ssandra.metrics.http;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.EpollServerSocketChannel;
+import io.netty.handler.logging.LogLevel;
+import io.netty.handler.logging.LoggingHandler;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+
+import javax.net.ssl.SSLException;
+import java.security.cert.CertificateException;
+
+public class NettyMetricsHttpServer {
+
+    public void start(EventLoopGroup group) {
+        // Configure SSL.
+//        final SslContext sslCtx = buildSslContext();
```

Comment:
Vector is loaded as a sidecar, and I think we can restrict metrics exposition to localhost only.
Maybe that would secure things enough that we don't have to deal with this?
If we implement this, can we re-use the Cassandra encryption settings?

Reply:
Vector isn't the only scraper, so listening only in localhost isn't a good solution.

We could secure this with kube-rbac-proxy or similar, we have examples / settings in our kustomize templates already.